.github/workflows/linter.yml

```diff
@@ -27,9 +29,11 @@ jobs:
             ${{ steps.pip-cache.outputs.dir }}
             /opt/hostedtoolcache/Python
             /Users/runner/hostedtoolcache/Python
-          key: ${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-pip-${{ hashFiles('**/setup.py') }}
+          key: ${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-pip-${{ hashFiles(format('**/py{0}-ci-requirements.txt', env.PYTHON)) }}
           restore-keys: |
             ${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-pip-
+      - name: Install pip-tools
+        run: pip install pip-tools
       - name: Install dependencies
         run: make install-python-ci-dependencies
```

Comment:
Any reason `pip-tools` can't be installed in the same step as `make install-python-ci-dependencies`? Applies to all the workflows that have been changed.

Reply:
No special reason actually. Is it better to change the step like this?

```yaml
- name: Install dependencies
  run: |
    pip install pip-tools
    make install-python-ci-dependencies
```

Reply:
No, I think that our `CI_REQUIRED` dependencies already has an entry for `pip-tools`, so I wonder if we even need to install `pip-tools` separately? `make install-python-ci-dependencies` should use the `py3.*-ci-requirements.txt` which already has an entry for `pip-tools`, so i think it may not be necessary.

Reply:
I see you point, but `pip-tools` must be installed before `make install-python` and `make install-python-ci-dependencies` are executed because `piptools sync` is executed in those Makefile targets.
`pip-tools` entry in requirements is not required any more, so I think it should be deleted.

The other approach is to use `pip install -r` instead of `piptools sync` in those targets. In this approach, installing `pip-tools` step can be deleted but we can't get the benefit of `piptools sync`. `pip install -r` executes only installation, but `piptools sync` installs/upgrades/uninstalls everything necessary to match the requirements.txt contents. See https://github.com/jazzband/pip-tools#example-usage-for-pip-sync for more info on `piptools sync`.

Reply:
Okay, I see your point - I'm fine with the way things are in the current version of your diff in that case. 

Final question, do you think we should pin the version of `pip-tools` we install in the github workflows?

Reply:
IMO, it's not required. I think newer the better.